keps/sig-cli/kustomize-subcommand-integration.md

```diff
@@ -0,0 +1,387 @@
+
+```
+apiVersion: v1
+data:
+JAVA_HOME: /opt/java/jdk
+JAVA_TOOL_OPTIONS: -agentlib:hprof
+kind: ConfigMap
+metadata:
+annotations:
+  oncallPager: 800-555-1212
+name: alices-myJavaServerEnvVars-7bc9c27cmf
+---
+apiVersion: v1
+data:
+password: c2VjcmV0Cg==
+username: YWRtaW4K
+kind: Secret
+metadata:
+annotations:
+  oncallPager: 800-555-1212
+name: alices-app-sec-c7c5tbh526
+type: Opaque
+```
+
+### Implementation Details/Notes/Constraints
+
+In contrast to `kubectl apply`, which was developed directly in kubernetes/kubernetes and had minimal usage or
+real world experience prior, kustomize was built as an outside of kubernetes/kubernetes in the
+kubernetes-sigs/kustomize repo.  After `kubectl apply` was published, many issues were uncovered in it that should
+have been discovered earlier.  Implementing kustomize independently allowed more time for gathering feedback and
+identifying issues.
+
+Kustomize library code will be moved from its current repository (location) to the cli-runtime repository used by
+kubectl.
+
+### Risks and Mitigations
+
+Low:
+
+- Kustomize can be run against remote urls.  A user could run it on a URL containing malicious workflows.  However this
+ would only generate the config, and the user would need to pipe it to apply for the workloads to be run.  This is also
+ true for `kubectl apply -f <url>` and or `kubectl create deployment --image <bad image>`.
+
+Low:
+
+- Kustomize has other porcelain commands to facilitate common workflows.  This proposal does not include integrating
+  them into kubectl.  Users would need to download kustomize separate to get these benefits.
+  
+Low:
+
+- `kubectl kustomize <dir>` doesn't take a `-f` flag like the other commands.
+
+## Graduation Criteria
+
+The API version for kustomize is defined in the kustomization.yaml file.  The KEP is targeted `v1beta1`.
+
+The criteria for graduating from `v1beta1` for the kustomize sub-command should be determined as part of
+evaluating the success and maturity of kustomize as a command within kubectl.
+
+Metrics for success and adoption could include but are not limited to:
+
+- number of kustomization.yaml files seen on sources such as GitHub
+- complexity (required) of kustomization.yaml files seen on sources such as GitHub.
+- (if available) number of calls to `kubectl kustomize` being performed
+- adoption or integration of kustomize by other tools
+
+Metrics for maturity and stability could include but are not limited to:
+
+- number and severity of kustomize related bugs filed that are intended to be fixed
+- the frequency of API changes and additions
+- understanding of relative use and importance of kustomize features
+
+**Note:** Being integrated into ResourceBuilder is *not* considered graduation and *not* gated on GA.
+
+## Implementation History
+
+Most implementation will be in cli-runtime
+
+- [ ] vendor `kustomize/pkg` into kubernetes
+- [ ] copy `kustomize/k8sdeps` into cli-runtime
```

Comment:
Yeah, at the end of the day I'm hoping that `cli-runtime` will contain the core of kustomize, but everything above (commands, etc) should exist outside `cli-runtime`. Maybe worth putting that here.